samples/MegaDrone/src/main/cpp/MegaDroneEngine.cpp

```diff
@@ -67,3 +68,19 @@ void MegaDroneEngine::setThreadAffinity() {
 
     mIsThreadAffinitySet = true;
 }
+
+oboe::Result MegaDroneEngine::createPlaybackStream(oboe::AudioStreamBuilder builder) {
+    oboe::Result result = builder.setSharingMode(oboe::SharingMode::Exclusive)
+        ->setPerformanceMode(oboe::PerformanceMode::LowLatency)
+        ->setFormat(oboe::AudioFormat::Float)
+        ->setCallback(mCallback.get())
+        ->openManagedStream(mStream);
+    return result;
+}
+
+void MegaDroneEngine::restartStream() {
+    createPlaybackStream(oboe::AudioStreamBuilder());
+    mAudioSource =  std::make_shared<Synth>(mStream->getSampleRate(), mStream->getChannelCount());
+    mCallback->setSource(std::dynamic_pointer_cast<IRenderableAudio>(mAudioSource));
```

Comment:
Excuse my ignorance by why is `std::dynamic_pointer_cast<IRendereableAudio>(mAudiosource)` better than just using `mAudioSource` (which is already an `IRenderableAudio`)?